galaxy_ng/app/tasks/promotion.py

```diff
@@ -1,92 +1,88 @@
+from django.conf import settings
+from django.utils.translation import gettext_lazy as _
+
 from pulpcore.plugin.tasking import add_and_remove, dispatch
-from pulp_ansible.app.models import CollectionVersionSignature
-from pulpcore.plugin.models import TaskGroup
+from pulp_ansible.app.models import AnsibleRepository
+from pulp_ansible.app.tasks.signature import sign
+from pulp_ansible.app.tasks.copy import move_collection
 
+from pulpcore.plugin.models import TaskGroup, SigningService
 
-def call_add_content_task(collection_version, repo):
-    """Dispatches the add content task
 
-    This is a wrapper to copy_content task
-    """
+SIGNING_SERVICE_NAME = settings.get("GALAXY_COLLECTION_SIGNING_SERVICE", "ansible-default")
+AUTO_SIGN = settings.get("GALAXY_AUTO_SIGN_COLLECTIONS", False)
+
+
+def auto_approve(src_repo_pk, cv_pk):
+    published_repos = AnsibleRepository.objects.filter(pulp_labels__pipeline="approved")
+    published_pks = list(published_repos.values_list("pk", flat=True))
+    staging_repo = AnsibleRepository.objects.get(pk=src_repo_pk)
+
+    add_and_remove(
+        src_repo_pk,
+        add_content_units=[cv_pk],
+        remove_content_units=[],
+    )
+
+    try:
+        signing_service = AUTO_SIGN and SigningService.objects.get(name=SIGNING_SERVICE_NAME)
+    except SigningService.DoesNotExist:
+        raise RuntimeError(_('Signing %s service not found') % SIGNING_SERVICE_NAME)
```

Comment:
I think it is better to translate only user facing messages (UI,API) and keep logs and errors untranslated so it is better to find when troubleshooting.

```suggestion
        raise RuntimeError('Signing %s service not found' % SIGNING_SERVICE_NAME)
```